core/shared/src/main/scala/fs2/Chunk.scala

```diff
@@ -1466,8 +1471,13 @@ object Chunk extends CollectorK[Chunk] {
   def concatBooleans(chunks: GSeq[Chunk[Boolean]]): Chunk[Boolean] =
     concatBooleans(chunks, chunks.foldLeft(0)(_ + _.size))
 
+  /**
+    * Concatenates the specified sequence of boolean chunks in to a single chunk.
+    * The `totalSize` parameter must be equal to the sum of the size of each chunk or
+    * otherwise an exception may be thrown.
+    */
   def concatBooleans(chunks: GSeq[Chunk[Boolean]], totalSize: Int): Chunk[Boolean] =
-    if (chunks.isEmpty) Chunk.empty
+    if (totalSize == 0) Chunk.empty
```

Comment:
the validation check / code pattern repeats itself a lot, does it make sense to have a HOF to wrap each of the concats in?

- totalSize check is only to 0, any need to check a negative value?
- are there tests on this?  (this is totes a newb question)

Reply:
Yeah I don't know how ugly you consider ClassTag or the cost of this indirection but you could dedupe all these methods with this private method:

```scala
  def concatBooleans(chunks: GSeq[Chunk[Boolean]], totalSize: Int): Chunk[Boolean] =
    concatGen[Boolean](chunks, totalSize, Chunk.booleans)
  
  private def concatGen[F: ClassTag](chunks: GSeq[Chunk[F]], totalSize: Int, chunkBuilder: Array[F] => Chunk[F]): Chunk[F] =
    if (chunks.isEmpty) Chunk.empty
    else {
      val arr = new Array[F](totalSize)
      var offset = 0
      chunks.foreach { c =>
        if (!c.isEmpty) {
          c.copyToArray(arr, offset)
          offset += c.size
        }
      }
      chunkBuilder(arr)
    }
```

Reply:
No need to check totalSize as it's passed to `new Array(..)` and that will handle it.

Reply:
I consolidated those primitive methods to a single private concatUnboxed